### PR TITLE
Updating CodeCoverage and deterministic build

### PR DESCRIPTION
**Pull request by eddynaka**
## Changes
- moved from `coverlet.collector` to `Microsoft.CodeCoverage`, with this change we will have Instrumentation.AspNet code coverage available
- added tags to generate a full `Deterministic` package
- maintained some portions commented, because probably in the future `coverlet` will support NETFramework projects entirely

### CODE REVIEW

File: build/process-codecoverage.ps1 [hunk 1]
```diff
@@ -0,0 +1,8 @@
+﻿$files = Get-ChildItem "TestResults" -Filter "*.coverage" -Recurse
+Write-Host $env:USERPROFILE
+foreach ($file in $files)
+{
+    $command = $env:USERPROFILE+ '\.nuget\packages\microsoft.codecoverage\16.6.1\build\netstandard1.0\CodeCoverage\CodeCoverage.exe analyze /output:' + $file.DirectoryName + '\' + $file.Name + '.xml '+ $file.FullName
+    Write-Host $command
+    Invoke-Expression $command
+}
```
Comment (reyang) [inline build/process-codecoverage.ps1:8]: nit: add a blank line

File: .github/workflows/dotnet-core-cov.yml [hunk 2]
```diff
@@ -35,14 +35,21 @@ jobs:
...
       run: dotnet build --configuration Release --no-restore
 
+#    - name: dotnet test
+#      run: dotnet test --collect:"XPlat Code Coverage" --results-directory:"TestResults" --configuration Release --no-build -- RunConfiguration.DisableAppDomain=true
+
     - name: dotnet test
-      run: dotnet test --collect:"XPlat Code Coverage" --results-directory:"TestResults" --configuration Release --no-build -- RunConfiguration.DisableAppDomain=true
+      run: dotnet test --collect:"Code Coverage" --results-directory:"TestResults" --configuration Release --no-build -- RunConfiguration.DisableAppDomain=true
+
+    - name: process code coverage
```
Comment (reyang) [inline .github/workflows/dotnet-core-cov.yml:44]:
```suggestion
    - name: Process code coverage
```

**Comment by eddynaka** [comment]
> LGTM, but I really have no idea what I'm looking at 🤣

LOL. I can explain!

right now, we are using coverlet.collector to get the data to generate code coverage. But this one doesn't work very well when we analyze .NET Framework tests. So, i'm moving to the same tool that VS uses.

The second part about deterministic thing, when we enable sourcelink, we can make the packages deterministic, which means that the path's won't point to a specific folder.

image
image

**Comment by roederja2** [comment]
Do we really want the release build to depend on Microsoft.CodeCoverage ? This is causing me some dependency conflicts to do with the TestSdk. Should the condition be to only use it for non release builds? https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/build/Common.props#L59

**Comment by eddynaka** [comment]
@roederja2 , open a new issue showing what is happening.

Even if we add the CodeCoverage, it will not be shipped.

**Comment by roederja2** [comment]
My issue may be related to a weird combination of beta builds - but I am wondering why there is a conditional reference for Release builds when I would have expected it to be the other way round.

For completeness this is what I'm seeing:

`Error	NU1107	Version conflict detected for Microsoft.CodeCoverage. Install/reference Microsoft.CodeCoverage 16.9.4 directly to project ExecutionServicesTests to resolve this issue. 
 MyTests -> Microsoft.NET.Test.Sdk 16.9.4 -> Microsoft.CodeCoverage (>= 16.9.4) 
 MyTests -> MyLibrary1 4.40.0 -> MyLibrary2 4.40.0 -> MyLibrary3 4.40.0 -> MyLibrary4 4.36.0 -> OpenTelemetry 1.0.2-alpha.0.16 -> Microsoft.CodeCoverage (= 16.9.0-preview-20201020-06).`

**Comment by eddynaka** [comment]
@roederja2 , you would only face this issue if you are adding the reference to opentelemetry project instead of opentelemetry nuget.

About being used in release instead of debug. no strong opinion.
Better to open a new issue and explain the use case.

**Pull request merged**